File: Storage/Code/278.第一个错误的版本.cpp
/*
 * @lc app=leetcode.cn id=278 lang=cpp
 *
 * [278] 第一个错误的版本
 */

// @lc code=start
// The API isBadVersion is defined for you.
// bool isBadVersion(int version);

class Solution {
public:
    int firstBadVersion(int n) {
        int left=1, right=n;
        while (left<right){
            int mid=left+((right-left)>>1);
            if(isBadVersion(mid)==false){
                left=mid+1;
            }else{
                right=mid;
            }
        }
        return left;
    }
};
// @lc code=end

File: Storage/Code/278.第一个错误的版本.py
#
# @lc app=leetcode.cn id=278 lang=python3
#
# [278] 第一个错误的版本
#

# @lc code=start
# The isBadVersion API is already defined for you.
# @param version, an integer
# @return an integer
# def isBadVersion(version):

class Solution:
    def firstBadVersion(self, n):
        """
        :type n: int
        :rtype: int
        """
        # 第一个返回true的版本
        left, right=1, n
        while left<right:
            mid=int(left+(right-left)/2)
            if isBadVersion(mid)==False:
                left=mid+1
            else:
                right=mid
        return left
        
# @lc code=end

